AlphaWallet/Transfer/ViewControllers/SendViewController.swift

```diff
@@ -357,16 +357,24 @@ class SendViewController: UIViewController, CanScanQRCode {
     }
 
     func didScanQRCode(_ result: String) {
-        activateAmountView()
-
-        guard let result = QRCodeValueParser.from(string: result) else { return }
-        switch result {
-        case .address(let recipient):
-            guard let tokenObject = storage.token(forContract: viewModel.transferType.contract) else { return }
-            let amountAsIntWithDecimals = EtherNumberFormatter.full.number(from: amountTextField.ethCost, decimals: tokenObject.decimals)
-            configureFor(contract: transferType.contract, recipient: .address(recipient), amount: amountAsIntWithDecimals)
-        case .eip681(let protocolName, let address, let functionName, let params):
-            checkAndFillEIP681Details(protocolName: protocolName, address: address, functionName: functionName, params: params)
+        if let result = QRCodeValueParser.from(string: result) {
```

Comment:
I don't think we should check the error here. So we should still guard and return like before the PR.

It's actually when fetching the token's details (symbol, decimals etc) where it fails if the contract is not a token. The error here could for example be if the EIP681 is not well-formed or maybe not even a URL.

It's the body of:

> self.delegate?.lookup(contract: contract, in: self) { data in

Around https://github.com/AlphaWallet/alpha-wallet-ios/pull/1876/files#diff-6aa3b98b96c470b61f3e704b8b9be6c8R396

Can you see if you can reproduce it?

Reply:
> I don't think we should check the error here. So we should still guard and return like before the PR.
> 
> It's actually when fetching the token's details (symbol, decimals etc) where it fails if the contract is not a token. The error here could for example be if the EIP681 is not well-formed or maybe not even a URL.
> 
> It's the body of:
> 
> > self.delegate?.lookup(contract: contract, in: self) { data in
> 
> Around https://github.com/AlphaWallet/alpha-wallet-ios/pull/1876/files#diff-6aa3b98b96c470b61f3e704b8b9be6c8R396
> 
> Can you see if you can reproduce it?

@hboon so you mean to show alert for all these cases
.name, .symbol, .balance, .decimals, .nonFungibleTokenComplete, .delegateTokenComplete, .failed:
in checkAndFillEIP681Details?

Reply:
@hboon can you confirm this?

Reply:
> so you mean to show alert for all these cases .name, .symbol, .balance, .decimals, .nonFungibleTokenComplete, .delegateTokenComplete, .failed: in checkAndFillEIP681Details?

Not for all of them.

Those like `.name`, etc are partial results. Ignore those

I think just:

* `.nonFungibleTokenComplete`
* `.delegateTokenComplete`
* but not `.failed`.

This should help to test:

1. Comment out this line in `SendViewController` (around line 395) and replace with `if false` to force it go to `else`:

```
if self.storage.token(forContract: contract) != nil {
```

Then:

A. Generate a QR code for a EIP681 link which should show the alert we want to show with this PR:

contract: 0x007bEe82BDd9e866b2bd114780a47f2261C684E3 (this is a wallet, so we expect it to fail)
receiver: 0x007bEe82BDd9e866b2bd114780a47f2261C684E3
amount: 1

B. Generate a QR code for a EIP681 link which should work and not show errors:

contract: 0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359 (this is the SAI token, on mainnet)
receiver: 0x007bEe82BDd9e866b2bd114780a47f2261C684E3
amount: 3000000000000000000

C. Generate a QR code for EIP681 transfer (click "ETH payment" in the page instead of ERC20) and make sure it doesn't trigger this error

Reply:
@hboon 
i walked through these cases, A and C works as you mentioned, but B doesnt work, (maybe i did something wrong)  all the time it enters to delegateTokenComplete case

Reply:
> i walked through these cases, A and C works as you mentioned, but B doesnt work, (maybe i did something wrong) all the time it enters to delegateTokenComplete case

@vladyslav-iosdev B works for me. Did you try on mainnet?